### ISSUE TITLE
uninitialized constant Karafka::BaseConsumer

### ISSUE THREAD

**Issue by mozillo** [karafka/karafka#436]
In Gemfile file
```
gem 'karafka', '~> 1.1.0'
gem 'ruby-kafka', '~> 0.6.8'
```

In my code
```
ApplicationConsumer = Class.new(Karafka::BaseConsumer)

class PdfConsumer < ApplicationConsumer
  def consume
    Rails.logger.info params
  end
end
```

Got errors
```
/Users/mot/projects/p1_api/app/responders/pdf_consumer.rb:1:in `<top (required)>': uninitialized constant Karafka::BaseConsumer (NameError)
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/activesupport-5.1.6.1/lib/active_support/dependencies/interlock.rb:12:in `block in loading'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/activesupport-5.1.6.1/lib/active_support/concurrency/share_lock.rb:149:in `exclusive'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/activesupport-5.1.6.1/lib/active_support/dependencies/interlock.rb:11:in `loading'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/railties-5.1.6.1/lib/rails/engine.rb:476:in `block (2 levels) in eager_load!'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/railties-5.1.6.1/lib/rails/engine.rb:475:in `each'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/railties-5.1.6.1/lib/rails/engine.rb:475:in `block in eager_load!'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/railties-5.1.6.1/lib/rails/engine.rb:473:in `each'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/railties-5.1.6.1/lib/rails/engine.rb:473:in `eager_load!'
	from /Users/mot/projects/p1_api/karafka.rb:6:in `<top (required)>'
	from /Users/mot/projects/p1_api/config/environment.rb:7:in `<top (required)>'
	from config.ru:3:in `require_relative'
	from config.ru:3:in `block in <main>'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/rack-2.0.6/lib/rack/builder.rb:55:in `instance_eval'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/rack-2.0.6/lib/rack/builder.rb:55:in `initialize'
	from config.ru:in `new'
	from config.ru:in `<main>'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/rack-2.0.6/lib/rack/builder.rb:49:in `eval'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/rack-2.0.6/lib/rack/builder.rb:49:in `new_from_string'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/rack-2.0.6/lib/rack/builder.rb:40:in `parse_file'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/rack-2.0.6/lib/rack/server.rb:319:in `build_app_and_options_from_config'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/rack-2.0.6/lib/rack/server.rb:219:in `app'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/railties-5.1.6.1/lib/rails/commands/server/server_command.rb:24:in `app'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/rack-2.0.6/lib/rack/server.rb:354:in `wrapped_app'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/railties-5.1.6.1/lib/rails/commands/server/server_command.rb:80:in `log_to_stdout'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/railties-5.1.6.1/lib/rails/commands/server/server_command.rb:42:in `start'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/railties-5.1.6.1/lib/rails/commands/server/server_command.rb:135:in `block in perform'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/railties-5.1.6.1/lib/rails/commands/server/server_command.rb:130:in `tap'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/railties-5.1.6.1/lib/rails/commands/server/server_command.rb:130:in `perform'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/railties-5.1.6.1/lib/rails/command/base.rb:63:in `perform'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/railties-5.1.6.1/lib/rails/command.rb:44:in `invoke'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/railties-5.1.6.1/lib/rails/commands.rb:16:in `<top (required)>'
	from /Users/mot/projects/p1_api/bin/rails:9:in `require'
	from /Users/mot/projects/p1_api/bin/rails:9:in `<top (required)>'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/spring-2.0.2/lib/spring/client/rails.rb:28:in `load'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/spring-2.0.2/lib/spring/client/rails.rb:28:in `call'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/spring-2.0.2/lib/spring/client/command.rb:7:in `call'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/spring-2.0.2/lib/spring/client.rb:30:in `run'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/spring-2.0.2/bin/spring:49:in `<top (required)>'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/spring-2.0.2/lib/spring/binstub.rb:31:in `load'
	from /Users/mot/.rvm/gems/ruby-2.3.1/gems/spring-2.0.2/lib/spring/binstub.rb:31:in `<top (required)>'
	from /Users/mot/projects/p1_api/bin/spring:15:in `require'
	from /Users/mot/projects/p1_api/bin/spring:15:in `<top (required)>'
	from bin/rails:3:in `load'
	from bin/rails:3:in `<main>'
```

**Comment by mensfeld** [comment]
@mozillo if it was resolved, is it still na issue? `1.1.0` is outdated and you should use the `1.2.*` releases.

**Comment by mensfeld** [comment]
Closing as it is not a bug but an upgrade change stated in the release notes and upgrade notes.
